gcp_variant_transforms/transforms/infer_headers.py

```diff
@@ -78,16 +90,109 @@ def _can_cast_to(self, value, cast_type):
     except ValueError:
       return False
 
-  def _infer_undefined_info_fields(self, variant, defined_headers):
+  def _get_corrected_type(self, defined_type, value):
+    # type: (str, Any) -> str
+    """Returns the corrected type according to `defined_type` and `value`.
+
+    It handles one special case PyVCF cannot handle, i.e., the defined type is
+    `Integer`, but the provided value is float. In this case, correct the type
+    to be `Float`.
+
+    Note that if `value` is a float instance but with an integer value
+    (e.g. 2.0), the type will stay the same as `defined_type`.
+    """
+    if defined_type == _HeaderTypeConstants.INTEGER:
+      if isinstance(value, float) and not float(value).is_integer():
```

Comment:
nit: this can just be `value.is_integer()` as you have already verified that it's a float.
P.S. Nice find on the is_integer method! did not know about it :)

Reply:
Ah, good point! Thanks.